### DISCUSSION TITLE
How to assert a schema follows an existing interface?

### DISCUSSION THREAD

**Discussion by Garrett-R** [colinhacks/zod#2559] [Q&A]
I also posted [on Stack Overflow](https://stackoverflow.com/questions/76617498/zod-equivalent-of-yups-schemaof).

In Yup, you can have code like this:

```typescript
import * as yup from "yup";

interface Dog {
  name: string;
}

const DogSchema: yup.SchemaOf<Dog> = yup.object({
  name: yup.string()
});
```

Here, `yup.SchemaOf` enforces that `DogSchema` and `Dog` are in agreement.

Any way to do that in Zod?

(I'm aware that, in most cases, using `yup.InferType` or `z.infer` is a more concise way to do this)

**Comment by fatguyintech** [comment]
There are workarounds but none built in.
